### PR TITLE
[NUI](a)Provide APIs to access sub-components

### PR DESCRIPTION
**Pull request by zhouleonlei**
### Description of Change ###
(a) Provide APIs to access sub-components of the component (e.g. Button.ButtonIcon).

### CODE REVIEW

**Comment by AchoWang** [comment]
Looks good to me.

File: src/Tizen.NUI.Components/Controls/Toast.cs [hunk 1]
```diff
@@ -117,6 +150,34 @@ public Toast(string style) : base(style)
             Initialize();
         }
 
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public TextLabel Textlabel
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/Toast.cs:154]: How about Textlabel -> TextLabel ?

File: src/Tizen.NUI.Components/Controls/Toast.cs [hunk 2]
```diff
@@ -117,6 +150,34 @@ public Toast(string style) : base(style)
...
+                        HeightResizePolicy = ResizePolicyType.UseNaturalSize,
+                        HorizontalAlignment = HorizontalAlignment.Center,
+                        VerticalAlignment = VerticalAlignment.Center,
+                        TextColor = Color.White
+                    };
+                    Add(textLabel);
+                }
+                return textLabel;
+            }
+            set
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/Toast.cs:175]: Can you make it internal API?
I think we should not allow users to replace sub-components of a components for now. It can cause unexpected behaviors.

File: src/Tizen.NUI.Components/Controls/Button.cs [hunk 3]
```diff
@@ -81,70 +84,67 @@ public class Button : Control
...
-        [EditorBrowsable(EditorBrowsableState.Never)]
-        protected ImageView ButtonIcon { get; set; }
+        private StringSelector translatableTextSelector = new StringSelector();
+        private StringSelector textSelector = new StringSelector();
+        private ColorSelector textColorSelector = new ColorSelector();
+        private FloatSelector pointSizeSelector = new FloatSelector();
+        private StringSelector iconURLSelector = new StringSelector();
+        private IconOrientation iconRelativeOrientation = IconOrientation.Left;
+        private Extents iconPadding;
+        private Extents textPadding;
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/Button.cs:147]: I think if there are same properties in Style, we can use them instead declaring them in Button. (to reduce size)
What do you think?

File: src/Tizen.NUI.Components/Controls/Button.cs [hunk 4]
```diff
@@ -277,55 +279,133 @@ public enum IconOrientation
...
         {
             get
             {
-                return (bool)GetValue(IsSelectableProperty);
+                return ButtonText.Text;
             }
             set
             {
-                SetValue(IsSelectableProperty, value);
+                ButtonText.Text = value;
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/Button.cs:392]: I think this patch (a) still has binding with corresponding Style, so following code should work.

```
Button button = new Button();
button.Text = "ABC";
Tizen.Log.Info("NUI", button.Style.Text.Text); //-> ABC
```

Comment (rabbitfor) [reply src/Tizen.NUI.Components/Controls/Button.cs:392]: All properties in components should work same as before.

File: src/Tizen.NUI.Components/Controls/Button.cs (hunk 3 above)
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/Button.cs:147]: I think if there are same properties in Style, we can use them instead declaring here to reduce size.

File: src/Tizen.NUI.Components/Controls/ButtonGroup.cs [hunk 5]
```diff
@@ -27,7 +27,7 @@ namespace Tizen.NUI.Components
     /// </summary>
     /// <since_tizen> 3 </since_tizen>
     [EditorBrowsable(EditorBrowsableState.Never)]
-    public class ButtonGroup : BindableObject, global::System.IDisposable
+    public class ButtonGroup : View
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/ButtonGroup.cs:30]: Can you explain why did you change inheritancy of ButtonGroup in this patch?
I think it is not related this patch.

File: src/Tizen.NUI.Components/Controls/Control.cs [hunk 6]
```diff
@@ -95,9 +115,20 @@ public Control(string styleSheet) : base()
             Initialize(style);
         }
 
-        internal void ApplyAttributes(View view, ViewStyle viewStyle)
+        /// Internal used.
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public ICommand Command
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/Control.cs:120]: What is the `Command`?

File: src/Tizen.NUI.Components/Controls/FlexibleView/FlexibleView.cs [hunk 7]
```diff
@@ -654,11 +654,11 @@ private void ShowScrollBar(uint millisecond = 700, bool flagAni = false)
             }
             if (mScrollBar.Direction == ScrollBar.DirectionType.Vertical)
             {
-                mScrollBar.Style.Thumb.Size = new Size(thickness, length);
+                mScrollBar.ThumbImage.Size = new Size(thickness, length);
```
Comment (rabbitfor) [inline src/Tizen.NUI.Components/Controls/FlexibleView/FlexibleView.cs:657]: If you change this, getting value from the Style will change.

File: src/Tizen.NUI.Components/Controls/Control.cs (hunk 6 above)
Comment (zhouleonlei) [reply src/Tizen.NUI.Components/Controls/Control.cs:120]: It is a request from Mr.Song. I will remove it from this patch

File: src/Tizen.NUI.Components/Controls/Toast.cs (hunk 1 above)
Comment (zhouleonlei) [reply src/Tizen.NUI.Components/Controls/Toast.cs:154]: OK

File: src/Tizen.NUI.Components/Controls/ButtonGroup.cs (hunk 5 above)
Comment (zhouleonlei) [reply src/Tizen.NUI.Components/Controls/ButtonGroup.cs:30]: This modification is related to Layout. I will remove it from this patch

File: src/Tizen.NUI.Components/Controls/Button.cs (hunk 3 above)
Comment (zhouleonlei) [reply src/Tizen.NUI.Components/Controls/Button.cs:147]: OK I will modify

File: src/Tizen.NUI.Components/Controls/Button.cs (hunk 4 above)
Comment (zhouleonlei) [reply src/Tizen.NUI.Components/Controls/Button.cs:392]: OK. Some codes need to  revert
